### ISSUE TITLE
On Windows, git_repository_hashfile doesn't recognize an absolute path containing backslashes as part of the workdir

### ISSUE THREAD

**Issue by jorio** [libgit2/libgit2#6825]
### Reproduction steps

1. **On Windows**, set up a repo with core.autocrlf = true.
2. In the workdir, create a text file "hellocrlf.txt" that contains CRLF line endings.
3. Hash the file using the repo's filtering rules by passing a relative path to `git_repository_hashfile`:
    `git_repository_hashfile(&a, "hellocrlf.txt", GIT_OBJECT_BLOB, NULL);`
4. Hash the file again, this time with an absolute path that separates directories with **backslashes**:
    `git_repository_hashfile(&b, "C:\\...\\hellocrlf.txt", GIT_OBJECT_BLOB, NULL);`

### Expected behavior

Hashes `a` and `b` should come out equal. The repo's filtering rules should apply to both. The hash should appear as if the file contained Unix line endings.

### Actual behavior

Hashes `a` and `b` are different. The filters were not applied when producing `b`.

`git_repository_hashfile` does not recognize the absolute path in step 4 as being part of the workdir. So, it doesn't apply the repo's filtering rules.

If the absolute path contains **forward** slashes instead, everything works as expected.

**Note:** We discovered this as we were [adding support for `git_repository_hashfile` in pygit2](https://github.com/libgit2/pygit2/pull/1298). Our unit test passes "native" paths (with `\`) to libgit2.

### Version of libgit2 (release number or SHA1)

1.8.1

### Operating system(s) tested

Windows.

Other systems (that don't use backslashes as dirsep) are not affected.

**Comment by ethomson** [comment]
Indeed; there's either a code issue or a docs issue here that we should resolve. PRs welcome.
